src/package-modules.js

```diff
@@ -657,6 +689,11 @@ async function createMetaPackageFromRepoDir(instruction, package, release) {
     throw {message: `Unable find package name and in composer.json for metapackage ${release.ref} in ${package.dir}`}
   }
   version = release.version || release.dependencyVersions[name] || version || release.ref;
+  
+  // If no version is found and ref looks like a branch name, convert to dev-branch format
+  if (!release && !dependencyVersions[name] && !composerConfig.version && ref && !ref.match(/^v?\d+\.\d+/)) {
+    version = `dev-${ref}`;
```

Comment:
There's a logic/reference error here. I don't think `ref` exists anymore as of the refactoring. That should probably be `release.ref`, but the condition excludes that via `!release`. What's the proper condition and ref?